<launch>

  <arg name="map_frame_id" default="map"/>
  <arg name="map_size_x" default="25"/>
  <arg name="map_size_y" default="25"/>
  <arg name="map_size_z" default="15"/>
  <arg name="map_x_origin" default="0.0"/>
  <arg name="map_y_origin" default="0.0"/>
  <arg name="map_z_origin" default="0.0"/>

  <node pkg ="param_env" name ="structure_map" type ="structure_map" output = "screen">    
    <!-- this is the map init position, not the odom-->

    <param name="map/x_size"     value="$(arg map_size_x)" />
    <param name="map/y_size"     value="$(arg map_size_y)" />
    <param name="map/z_size"     value="$(arg map_size_z)" />
    <param name="map/x_origin"   value="$(arg map_x_origin)"/>
    <param name="map/y_origin"   value="$(arg map_y_origin)"/>
    <param name="map/z_origin"   value="$(arg map_z_origin)"/>

    <param name="map/add_noise"  value="false"/>
    <param name="map/resolution" value="0.2"/>        
    <param name="map/frame_id"     value="$(arg map_frame_id)" />

    <!-- total obstacle ratios -->
    <param name="map/cylinder_ratio" value="0.02" type="double"/>
    <param name="map/circle_ratio"   value="0.02" type="double"/>
    <param name="map/gate_ratio"     value="0.02" type="double"/>
    <param name="map/ellip_ratio"    value="0.01" type="double"/>
    <param name="map/poly_ratio"     value="0.01" type="double"/>

    <param name="params/w1" value="0.1"/>
    <param name="params/w2" value="0.6"/>          
    <param name="params/w3" value="1.2"/>
    <param name="params/w4" value="3.0"/>


    <param name="start_num" value="5080900"/>
  
    <param name="dataset_path" value="$(find param_env)/dataset/"/>


  </node>


  <!-- <arg name="rviz_config" default="$(find param_env)/launch/default.rviz"/>
  <node pkg="rviz"
    type="rviz"
    name="rviz"
    args="-d $(arg rviz_config)"/> -->


</launch>
